File: backend_store/src/main/java/com/ghiffaryr/store/enums/OrderStatusEnum.java
package com.ghiffaryr.store.enums;

public enum OrderStatusEnum implements CodeEnum {
    NEW(0, "New Order"),
    FINISHED(1, "Finished"),
    CANCELED(2, "Canceled")
    ;

    private int code;
    private String msg;

    OrderStatusEnum(Integer code, String msg) {
        this.code = code;
        this.msg = msg;
    }

    @Override
    public Integer getCode() {
        return code;
    }
}

File: frontend_web_store_vite_ts/src/enums/OrderStatusEnum.ts
import { MappedType } from '@/utils/type';

const OrderStatusEnum: MappedType<number, string> = {
  0: 'New',
  1: 'Finished',
  2: 'Canceled',
};

export enum EOrderStatus {
  'New' = 0,
  'Finished',
  'Canceled',
}

Object.freeze(OrderStatusEnum);

export default OrderStatusEnum;
